core/src/wallet/bitcoin/PersistentUTXOSource.cpp

```diff
@@ -0,0 +1,42 @@
+#include <bytes/serialization.hpp>
+#include <wallet/bitcoin/PersistentUTXOSource.hpp>
+#include <database/BlockchainDB.hpp>
+
+namespace ledger {
+    namespace core {
+        namespace bitcoin {
+            PersistentUTXOSource::PersistentUTXOSource(
+                const std::shared_ptr<db::BlockchainDB>& blockchainDB,
+                const std::shared_ptr<UTXOSourceInMemory>& inMemorySource
+            ) : _db(blockchainDB), _inMemorySource(inMemorySource) {
+            }
+
+            Future<UTXOSourceList> PersistentUTXOSource::getUTXOs(std::shared_ptr<api::ExecutionContext> ctx) {
+                auto self = shared_from_this();
+
+                return _inMemorySource->getUTXOs(ctx).template flatMap<UTXOSourceList>(ctx, [=](const UTXOSourceList& sourceList) {
+                    return _db->GetLastBlock().template map<UTXOSourceList>(ctx, [=](const Option<std::pair<uint32_t, db::ReadOnlyBlockchainDB::RawBlock>>& lastBlock) {
```

Comment:
Not your fault. Just thoughts - GetLastBlock read entire block, that can be big, actually it needs only height, maybe we need to remember last returned value, or add new method to DB (or have another db for this purposes, as you write in PR comment)

Reply:
Right, it is only height.
Go ahead if you fill like you are on it.

Reply:
Done in the current last commit.